BetaUtils.DotNetCore/Configuration/CoreConfigurationProvider.cs

```diff
@@ -0,0 +1,309 @@
+        /// This method will return an Guid value of a configuration item for provided key. 
+        /// If an item for the provided key does not exist, this method will return a provided default value
+        /// </summary>
+        /// <param items="key">Key of the configuration item</param>
+        /// <param items="defaultValue">Value that will be returned if configuration item with provided key does not exist</param>
+        /// <returns>Guid value of the configuration item</returns>
+        public Guid GetGuidValueOrDefault(string key, Guid defaultValue = default)
+        {
+            var stringValue = GetStringValue(key);
+
+            if (!Guid.TryParse(stringValue, out Guid result))
+            {
+                result = defaultValue;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// This method will return an integer value of a configuration item for provided key
+        /// </summary>
+        /// <param items="key">Key of the configuration item</param>
+        /// <returns>Int value of the configuration item</returns>
+        /// <exception cref="MissingConfigurationException">If item for provided key does not exist or it is not an integer</exception>
+        public int GetIntValue(string key)
+        {
+            var stringValue = GetStringValue(key);
+
+            VerifyIfConfigurationValueExists(key, stringValue);
+
+            if (int.TryParse(stringValue, out var result))
+            {
+                return result;
+            }
+
+            throw new MissingConfigurationException($"Configuration key '{key}' contains a value that is not an integer. The value supplied is '{stringValue}'.");
+        }
+
+        /// <summary>
+        /// This method will return an integer value of a configuration item for provided key. 
+        /// If an item for the provided key does not exist, this method will return a provided default value
+        /// </summary>
+        /// <param items="key">Key of the configuration item</param>
+        /// <param items="defaultValue">Value that will be returned if configuration item with provided key does not exist</param>
+        /// <returns>Int value of the configuration item</returns>
+        public int GetIntValueOrDefault(string key, int defaultValue = 0)
+        {
+            var stringValue = GetStringValue(key);
+
+            if (!int.TryParse(stringValue, out int result))
+            {
+                result = defaultValue;
+            }
+
+            return result;
+        }
+
+        /// <summary>
+        /// This method will return text value of a configuration item for provided key
+        /// </summary>
+        /// <param items="key">Key of the configuration item</param>
+        /// <returns>Text value of the configuration item</returns>
+        /// <exception cref="MissingConfigurationException">If item for provided key does not exist</exception>
+        public string GetStringValue(string key)
```

Comment:
```C# 
<Nullable>enable</Nullable>
```
As much as its pain could be useful in cases such as this one or more complex cases where  incoming value is not guaranteed to exist.
Passing null to this method from any source will produce unhandled ArgumentNullException.